### PR TITLE
SR11: Modified restaurant schema and repo for SR11 req

### PR DESCRIPTION
**Pull request by edgarspy**
Summary:
Implements the data layer for SR11 by introducing full CRUD support for restaurants and menu items through a new repository layer, updated schemas, and unit tests

Changes:
- Added restaurant_repo.py with full CRUD functions: create_restaurant, update_restaurant, delete_restaurant, add_menu_item, update_menu_item, delete_menu_item, get_active_menu_items, get_all_restaurants, reset_restaurants
- Migrated existing data functions into restaurant_repo.py
- Added RestaurantCreate, RestaurantUpdate, and MenuItemUpdate schemas to restaurant.py
- Added owner_id to restaurant records

Testing:
- Unit test: verifying validation rules for RestaurantCreate, RestaurantUpdate, MenuItemCreate, and MenuItemUpdate in test_restaurant_data_layer_schema.py
- Unit test: verifying all repo CRUD functions, including some edge cases like missing values in test_restaurant_data_layer_repo.py

### CODE REVIEW

File: backend/app/schemas/restaurant.py [hunk 1]
```diff
@@ -13,6 +13,20 @@ class RestaurantOut(BaseModel):
     opening_hours: str
     menu: List[MenuItemOut] = []
```
Comment (eleonoransella) [inline backend/app/schemas/restaurant.py]: Should menu use Field(default_factory=list) instead of [] to avoid using a shared default list?
